app/models/user.rb

```diff
@@ -456,6 +456,16 @@ def close_account!
     AccountDeletion.create(:person => self.person)
   end
 
+  def lock_account!
+    self.locked_at = Time.now
+    self.save
+  end
```

Comment:
See this line: https://github.com/diaspora/diaspora/blob/develop/app/views/admins/_user_entry.haml#L43

I'd say make that a bit more pretty by adding `closed_account?` to the User model, you can just delegate that call to the associated Person model, as you can see in the current usage, then replace the call I linked with the it and wrap the entire locking code into an `unless user.closed_account?`

I don't see why you would need an `access_unlocked?` method.